src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -419,7 +419,7 @@ SHAMapAddNode SHAMap::addRootNode (Blob const& rootNode,
 
     assert (seq_ >= 1);
     auto node = SHAMapAbstractNode::make(
-        rootNode, 0, format, uZero, false, f_.journal ());
+        rootNode, 0, format, SHAMapHash{uZero}, false, f_.journal ());
```

Comment:
Oh.  I'm learning something here.  I knew there was a compiler generated flag on function scoped statics, but I didn't know there was one on file scope statics (on line 29).  I thought there was (still) an access ordering problem if you try to access a file scoped static before `main()` is called.

I don't yet understand how the suggestion would make things worse.  We have an existing file scope static on line 29 that looks like this:
```
static const uint256 uZero;
```
We use that file scope static like this on lines 422, 463, and 541:
```
auto node = SHAMapAbstractNode::make(
    rootNode, 0, format, SHAMapHash{uZero}, false, f_.journal ());
```
I'm suggesting that we change the declaration of `uZero` on line 29 to be the type we want.  Like this:
```
static const SHAMapHash uZero {beast:zero};
```
then we can use it like this:
```
auto node = SHAMapAbstractNode::make(
    rootNode, 0, format, uZero, false, f_.journal ());
```
What did I miss?